app/controllers/users_controllers.rb

```diff
@@ -0,0 +1,35 @@
+class UsersControllers < ApplicationController
+  before_action :require_admin, only: [:index, :show, :new, :create, :destroy]
+  before_action :require_user, only: [:edit, :update]
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    authorize @user
```

Comment:
It is possible to do like that
@user = authorize(User.find())
But take a look at previous comment